soc/nxp/imxrt/imxrt118x/soc.c

```diff
@@ -56,6 +59,14 @@ static const clock_arm_pll_config_t armPllConfig_BOARD_BootClockRUN = {
 };
 #endif
 
+#if defined(CONFIG_WDT_MCUX_IMX_RTWDOG)
+#define RTWDOG_IF_SET_SRC(n, i)                                                                    \
+        if (strcmp(DT_PROP(DT_NODELABEL(rtwdog##n), status), "okay") == 0) {                       \
```

Comment:
Use `if (IS_ENABLED(DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(rtwdog#nn), nxp_imx_rtwdog, okay))` here, instead of the runtime check.

Reply:
Updated. Thanks Daniel for all these great advises.